mods/parsoid.js

```diff
@@ -115,19 +134,39 @@ PSP.saveParsoidResult = function (restbase, req, format, tid, parsoidResp) {
     }
 };
 
-PSP.generateAndSave = function(restbase, req, format, tid) {
+// Temporary work-around for Parsoid issue
+// https://phabricator.wikimedia.org/T93715
+function normalizeHtml(html) {
+    return html && html.toString
+        && html.toString().replace(/ about="[^"]+"(?=[\/> ])/g, '');
+}
+function sameHtml(a, b) {
+    return normalizeHtml(a) === normalizeHtml(b);
+}
+
+PSP.generateAndSave = function(restbase, req, format, currentContentRes) {
     var self = this;
-    if (!tid) {
-        throw new Error('no tid');
-    }
+    var tid = uuid.v1();
     // Try to generate HTML on the fly by calling Parsoid
     var rp = req.params;
+    var storageRequest = null;
+
     return restbase.get({
         uri: new URI([rp.domain,'sys','parsoid','pagebundle',
                      normalizeTitle(rp.title),rp.revision])
     })
-    .then(function(parsoidResp) {
-        return self.saveParsoidResult(restbase, req, format, tid, parsoidResp);
+    .then(function(res) {
+        if (format === 'html' && currentContentRes
+                && sameHtml(res.body.html.body, currentContentRes.body)) {
+            // New render is the same as the previous one, no need to store
+            // it.
+            //console.log('not saving a new revision!');
+            restbase.metrics.increment('sys_parsoid_generateAndSave.unchanged_rev_render');
+
+            return self.wrapContentReq(restbase, req, P.resolve(currentContentRes));
```

Comment:
That's pretty subtle (and the main save path also re-wraps). I kind of prefer to have the wrapping as late as possible, so it's clear from looking at the code that it happens.

I actually wonder if we should ever expose pagebundle publicly. Right now the only reason to have a separate entry point for it is to get separate stats, but with metrics now being a public property on restbase we could just create custom metrics for that in the module.